src/projections/spilhaus.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Implementation of the Spilhaus projections based on Adams World in a Square
+ * II.
+ *
+ * Explained in https://github.com/OSGeo/PROJ/issues/1851
+ *
+ * Copyright (c) 2025 Javier Jimenez Shaw
+ *
+ * Related material
+ * ----------------
+ *
+ * Map Projections - A Working Manual. 1987. John P. Snyder
+ * Sections 3 and 5.
+ * https://doi.org/10.3133/pp1395
+ * Online at https://neacsu.net/docs/geodesy/snyder/2-general/
+ */
+
+#include <errno.h>
+#include <math.h>
+
+#include "proj.h"
+#include "proj_internal.h"
+#include <math.h>
+
+C_NAMESPACE PJ *pj_adams_ws2(PJ *);
+
+PROJ_HEAD(spilhaus, "Spilhaus") "\n\tSph&Ell";
+
+namespace { // anonymous namespace
+struct pj_spilhaus_data {
+    double cosalpha;
+    double sinalpha;
+    double beta;
+    double lambda_0;
+    double conformal_distortion;
+    double cosrot;
+    double sinrot;
+    PJ *adams_ws2;
+};
+
+double conformal_lat(double lat, double e) {
```

Comment:
could we move that do a dedicated file in src/  ? (if it is not already there, but can't spot it)
I can see that mod_ster_setup() does this computation too and could be another user

Reply:
Probably a dedicated one. We have some files phi2.cpp, mlfn.cpp, msfn.cpp, qsfn.cpp, tsfn.cpp with "similar" things but whose naming is apparently meant to be obscure enough that I never know what they do...
Or maybe have a "latitudes.cpp" file where we would group together code to transform between geographic latitude and all those "other latitudes ? But that may be work for another pull request

Reply:
I created `latitudes.cpp`, and move the functions there.